autotest.pm

```diff
@@ -1,18 +1,6 @@
-# Copyright © 2009-2013 Bernhard M. Wiedemann
-# Copyright © 2012-2021 SUSE LLC
-#
-# This program is free software; you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation; either version 2 of the License, or
-# (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License along
-# with this program; if not, see <http://www.gnu.org/licenses/>.
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2012 SUSE LLC
```

Comment:
somehow you lost all the `-2021` for SUSE

Reply:
yes, thanks for noticing. That was the intention of https://github.com/os-autoinst/os-autoinst/pull/1800/commits/d0af9e9b143820311c0348383b54731c17701919 in particular. Please see the individual commit messages where I tried to provide sufficient detailed reasoning

Reply:
I disagree with that change. Even your googe link says adding the current year is fine on modifications. https://www.gnu.org/prep/maintain/html_node/Copyright-Notices.html explicity says to add the current year on non-trivial changes. And I doubt it's up to you to dedicde *now* what were non-trivial changes at the time.

Reply:
No problem, I updated https://github.com/os-autoinst/os-autoinst/pull/1799 accordingly and will update this PR later.

Reply:
Updated. I am not saying that the changes are non-trivial. Copying the commit description from the commit which now removed:
Commonly the copyright extends only if significant changes are applied to files
and even in that case renewal of the copyright is automatic based on the actual
content differences, not needing an explicit statement in the copyright header.
We usually extended the copyright with every new year as soon as changes have
been made. This has been seen as a nuisance by many and causes unnecessary
changes in the git log.

Other references like https://opensource.google/docs/copyright/ and
https://matija.suklje.name/how-and-why-to-properly-write-copyright-statements-in-your-code
and https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/ also
suggest to only include the original year of publication in the copyright
notice. In practice it would likely not matter if a 70 year period of copyright
is extended by just a handful of years anyway.

It is correct that gnu.org states that we should include subsequent years on changes:
> To update the list of year numbers, add each year in which you have made nontrivial changes to the package. (Here we assume you’re using a publicly accessible revision control server, so that every revision installed is also immediately and automatically published.)"

So technically I was proposing to not follow the GNU GPL recommendations but what I understood to be the current "legal practice".

Also interesting is the following suggestion from GNU GPL:
> When you add the new year, it is not required to keep track of which files have seen significant changes in the new year and which have not. It is recommended and simpler to add the new year to all files in the package, and be done with it for the rest of the year.

I understand this as: Use at best an automatic tool to update all files to the current year at the beginning of each year.

Reply:
That's what a lot of GNU projects do, yes. I don't propose that though